src/node/server.ts

```diff
@@ -1,74 +1,60 @@
 import { RequestListener, IncomingMessage, ServerResponse } from "http";
 import { TextDecoder } from "util";
-import { ReadableStream } from "web-streams-polyfill/ponyfill/es2018";
+import { once } from "events";
+import * as util from "util";
+import * as stream from "stream";
 
 import { HttpHandler } from "../core/http4ts";
 import { HttpRequest, HttpResponse } from "../core/http";
 import { HttpStatus } from "../core/http-status";
 import { HttpRequestImpl } from "../core/HttpRequestImpl";
-import { HttpBodyImpl } from "../core/HttpBodyImpl";
 import { setupEnvironment } from "../core/env";
-import { toReadableStream } from "./streamUtils";
+import { HttpBodyImpl } from "../core/HttpBodyImpl";
 
-async function translateRequest(
-  nodeReq: IncomingMessage
-): Promise<HttpRequest> {
+const finished = util.promisify(stream.finished);
+
+function toHttp4tsRequest(nodeReq: IncomingMessage): HttpRequest {
   return new HttpRequestImpl(
     nodeReq.url || "", // TODO: Maybe failing is better
-    new HttpBodyImpl(toReadableStream(nodeReq)),
+    new HttpBodyImpl(nodeReq),
     nodeReq.method || "", // TODO: Maybe failing is better
     nodeReq.headers
   );
 }
 
-async function readToEnd(
-  reader: ReadableStreamDefaultReader,
-  nodeRes: ServerResponse,
-  onFail: (reason: any) => void
+async function writeToNodeResponse(
+  http4tsResponse: HttpResponse,
+  nodeResponse: ServerResponse
 ) {
-  reader.read().then(({ done, value }) => {
-    if (done) {
-      nodeRes.end();
-      return;
-    }
-    nodeRes.write(value, onFail);
-    readToEnd(reader, nodeRes, onFail);
-  });
-}
+  nodeResponse.writeHead(http4tsResponse.status, http4tsResponse.headers);
 
-function writeResponse(
-  res: HttpResponse,
-  nodeRes: ServerResponse
-): Promise<void> {
-  return new Promise<void>(async (resolve, reject) => {
-    nodeRes.statusCode = res.status;
-    for (const key in res.headers) {
-      if (res.headers.hasOwnProperty(key)) {
-        const value = res.headers[key];
-        nodeRes.setHeader(key, value || "");
-      }
+  for await (const chunk of http4tsResponse.body) {
+    if (!nodeResponse.write(Buffer.from(chunk))) {
+      await once(nodeResponse, "drain");
     }
-    const reader = res.body.stream.getReader();
-    await readToEnd(reader, nodeRes, reject);
-    resolve();
-  });
+  }
+  nodeResponse.end();
+  await finished(nodeResponse);
```

Comment:
in order to understand this function, read [this](https://2ality.com/2019/11/nodejs-streams-async-iteration.html#writing-to-a-writable-stream-in-an-async-function).